### PR TITLE
Add support for -print0

### PR DESCRIPTION
**Pull request by refi64**
Signed-off-by: Ryan Gonzalez <ryan.gonzalez@collabora.com>

<hr>

I admittedly had this ready for a bit now, but I knew it would have merge conflicts with #120 so just waited until that was in first

### CODE REVIEW

File: src/find/matchers/printer.rs [hunk 1]
```diff
@@ -8,27 +8,46 @@ use walkdir::DirEntry;
...
 
+pub enum PrintDelimiter {
+    Newline,
+    Null,
+}
+
+impl std::fmt::Display for PrintDelimiter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PrintDelimiter::Newline => write!(f, "\n"),
```
Comment (sylvestre) [inline src/find/matchers/printer.rs:19]:
```suggestion
            PrintDelimiter::Newline => writeln!(f),
```

**Comment by refi64** [comment]
Fixed clippy now (really I should be used to running that already)

**Comment by sylvestre** [comment]
could you please rebase this patch to compare with the other testsuite? (I don't know why github isn't allowing this in the interface)

**Comment by sylvestre** [comment]
Nice:
`Warning: Congrats! The bfs test test_print0 is now passing!`

**Pull request merged**